### PR TITLE
Deployments tidy up [WIP]

### PR DESCRIPTION
**Pull request by ivandrofly**

### CODE REVIEW

**Comment by shiftkey** [comment]
There's an outstanding convention test failing here:

```
Octokit.Tests.Conventions.InterfaceMissingMethodsException : Methods not found on interface IObservableRepositoriesClient which are required:
       - get_Deployments
```

I'm not sure how I feel about the pluralization change on the whole, especially when we made a lot of changes to `IGitHubClient` to drop these...

**Comment by ryangribble** [comment]
Yeah, for better or worse the octokit convention is for singularly named clients. 

It's not really fitting to change just this one item to pluralised form

**Comment by ivandrofly** [comment]
Updated /cc @shiftkey | @ryangribble

File: Octokit/GitHubClient.cs [hunk 1]
```diff
@@ -247,15 +246,6 @@ public Uri BaseAddress
         /// </remarks>
         public ISearchClient Search { get; private set; }
 
-        // TODO: this should be under Repositories to align with the API docs
```
Comment (ivandrofly) [inline Octokit/GitHubClient.cs]: Deployment is already aligned with API docs

File: Octokit/Clients/RepositoriesClient.cs [hunk 2]
```diff
@@ -482,7 +482,7 @@ public Task<IReadOnlyList<Repository>> GetAllForOrg(string organization, ApiOpti
         /// Client for GitHub's Repository Deployments API
         /// </summary>
         /// <remarks>
-        /// See the <a href="http://developer.github.com/v3/repos/deployment/">Collaborators API documentation</a> for more details
```
Comment (ivandrofly) [inline Octokit/Clients/RepositoriesClient.cs]: Broken!!!

**Comment by ryangribble** [comment]
Interesting! It looks like DeploymentsClient was moved under Repository back in 2014 and removed from the `IGitHubClient` interface, but inadvertently left behind in the `GitHubClient` concrete class 😝 

Nice pickup!

> Hmm... looks like some changes are required in ObservableDeploymentStatusClient.cs too... WIP.

No worries, Ive marked `[WIP]` in the PR title - just edit that out when it's ready for review/merging

File: Octokit/Clients/RepositoriesClient.cs [hunk 3]
```diff
@@ -482,7 +482,7 @@ public Task<IReadOnlyList<Repository>> GetAllForOrg(string organization, ApiOpti
...
+        /// See the <a href="https://developer.github.com/v3/repos/deployments/">Collaborators API documentation</a> for more details
```
Comment (shiftkey) [inline Octokit/Clients/RepositoriesClient.cs]: It's also still called `Collaborators` in here due to someone copy-pasting things. Could you update that?

Comment (ivandrofly) [inline Octokit/Clients/RepositoriesClient.cs]: Sure ;)

**Comment by ryangribble** [comment]
Just getting back on top of things, looks like this one is good to go

Many thanks for the contribution!


LGTM

**Pull request merged**